server/moodle/blocks/homework/amd/src/homework_injector.js

```diff
@@ -50,14 +58,17 @@ define(function() {
                                     const homeworkLinkDiv = document.createElement('div');
                                     homeworkLinkDiv.className = 'description-content col-11';
                                     // Create the new string with all homework info for that course
-                                    let newString ='';
+                                    let newString = '';
                                     Object.values(filteredHomework).forEach((homeworkInfo) => {
-                                        newString += `<p>HomeWork name = ${homeworkInfo.name} Intro = ${homeworkInfo.intro} link = <a href="https://localhost/mod/homework/view.php?id=${homeworkInfo.id}">link to homework</a>` +
-                                            ` dueDate = ${homeworkInfo.duedate ? `Duedate is ${Date(homeworkInfo.duedate)}` : 'No duedate'}</p>`;
+                                        let convertedTime = new Date(homeworkInfo.duedate * 1000);
+                                        newString += `<p>HomeWork name = ${homeworkInfo.name} <br>
+                                                         Intro = ${homeworkInfo.intro ? homeworkInfo.intro : "No intro"} <br>
+                                                         link = <a href="${window.location.href.replace("my/", "mod/homework/view.php?id=")}${homeworkInfo.cmid}">link to homework</a></p>
```

Comment:
I stedet for at bruge `.replace("/my")`, så brug `href="/mod/homework/view.php?id=${homeworkInfo.course_module_id}"`